blueprints/third-party-solutions/wordpress/cloudrun/main.tf

```diff
@@ -0,0 +1,177 @@
+    "roles/logging.admin"                  = local.all_principals_iam
+    "roles/iam.serviceAccountUser"         = local.all_principals_iam
+    "roles/iam.serviceAccountTokenCreator" = local.all_principals_iam
+  }
+  cloud_sql_conf = {
+    database_version = "MYSQL_8_0"
+    tier             = "db-g1-small"
+    db               = "wp-mysql"
+    user             = "admin"
+    pass             = "password"
+  }
+  wp_user = "user"
+}
+
+
+# either create a project or set up the given one
+module "project" {
+  source          = "../../../../modules/project"
+  name            = var.project_id
+  parent          = try(var.project_create.parent, null)
+  billing_account = try(var.project_create.billing_account_id, null)
+  project_create  = var.project_create != null
+  prefix          = var.project_create == null ? null : var.prefix
+  iam             = var.project_create != null ? local.iam : {}
+  iam_additive    = var.project_create == null ? local.iam : {}
+  services = [
+    "run.googleapis.com",
+    "logging.googleapis.com",
+    "monitoring.googleapis.com",
+    "sqladmin.googleapis.com",
+    "sql-component.googleapis.com",
+    "vpcaccess.googleapis.com",
+    "servicenetworking.googleapis.com"
+  ]
+}
+
+
+resource "random_password" "wp_password" {
+  length = 8
+}
+
+
+# create the Cloud Run service
+module "cloud_run" {
+  source     = "../../../../modules/cloud-run"
+  project_id = module.project.project_id
+  name       = "${local.prefix}cr-wordpress"
+  region     = var.region
+
+  containers = [{
+    image = var.wordpress_image
+    ports = [{
+      name           = "http1"
+      protocol       = null
+      container_port = var.wordpress_port
+    }]
+    options = {
+      command  = null
+      args     = null
+      env_from = null
+      # set up the database connection
+      env = {
+        "APACHE_HTTP_PORT_NUMBER" : var.wordpress_port
+        "WORDPRESS_DATABASE_HOST" : module.cloudsql.ip
+        "WORDPRESS_DATABASE_NAME" : local.cloud_sql_conf.db
+        "WORDPRESS_DATABASE_USER" : local.cloud_sql_conf.user
+        "WORDPRESS_DATABASE_PASSWORD" : local.cloud_sql_conf.pass
+        "WORDPRESS_USERNAME" : local.wp_user
+        "WORDPRESS_PASSWORD" : random_password.wp_password.result
+      }
+    }
+    resources     = null
+    volume_mounts = null
+  }]
+
+  iam = {
+    "roles/run.invoker" : [var.cloud_run_invoker]
+  }
+
+  revision_annotations = {
+    autoscaling = {
+      min_scale = 1
+      max_scale = 2
+    }
+    # connect to CloudSQL
+    cloudsql_instances  = [module.cloudsql.connection_name]
+    vpcaccess_connector = null
+    # allow all traffic
+    vpcaccess_egress    = "all-traffic"
+  }
+  ingress_settings = "all"
+
+  # create a VPC connector for the ClouSQL VPC
+  vpc_connector_create = {
+    ip_cidr_range = var.connector_cidr
+    name          = "${local.prefix}wp-connector"
+    vpc_self_link = module.vpc.self_link
+  }
+}
+
+
+# create a VPC for CloudSQL
+module "vpc" {
```

Comment:
Can you move the VPC resources to a `vpc.tf` file? I would also try to create the connector here instead of having a requirement on it, if possible.

Reply:
Would it then maybe make sense to put all the CloudSQL resources (CloudSQL, VPC and FW) to a separate .tf file (like cloudsql.tf)? And for the connector, do you mean I should separately create a google_vpc_access_connector resource?

Reply:
> Would it then maybe make sense to put all the CloudSQL resources (CloudSQL, VPC and FW) to a separate .tf file (like cloudsql.tf)?

Sure, organize resources in file in a way that makes sense to you. As `main.tf` is pretty large, I just think we would improve readability by using additional files where it makes sense.

> And for the connector, do you mean I should separately create a google_vpc_access_connector resource?

Yep, that would be great and allow both use cases: where you don't have a connector, and where you already have one.

Reply:
OK, so I've restructured the files and added the connector resource, please take a look. If you want a count (like 1 or 0) for the connector creation, that would only make sense with additional changes for the CloudSQL, for example, a count there as well. Since we wanted to go with the easiest setup possible, I would not like to overcomplicate things here, this should be targeted at people having no setup at all. Is that OK for you?

Reply:
I think we should have a count, it's really simple to add. And yes, we could to the same for CloudSQL but it's less important as connector might be shared, while using a dedicated CloudSQL makes more sense.